src/server/pfs/server/driver.go

```diff
@@ -1346,7 +1354,7 @@ func (d *driver) deleteCommit(pachClient *client.APIClient, userCommit *pfs.Comm
 		}
 
 		// 3) Validate the commit (check that it has no provenance) and delete it
-		if len(userCommitInfo.Provenance) > 0 {
+		if provenantOnInput(userCommitInfo.Provenance) {
```

Comment:
Just want to double check that this is doing the right thing here.

Reply:
So doing this will make it so that people can delete commits that were emitted by a spout pipeline, which seems to make sense, since they're more like input commits than output commits.

It would be good to have a test where we delete a commit that was outputted by a spout to make sure this behavior works as we expect and doesn't break in the future.